### ISSUE TITLE
Prepare 9.3.0

### ISSUE THREAD

**Issue by tiagocoutinho** [tango-controls/pytango#255]
This is to inform the community that 9.3.0 is being prepared

- [x] Read steps in the how-to-contribute docs for making a release
- [x] Edit the changelog
- [x] Bump the version
- [x] Update `.travis.yml` with a conda package using a final cppTango 9.3.x release
- [x] Merge develop into stable
- [x] Make sure Travis is OK on stable branch
- [x] Make sure the documentation is updated (readthedocs)
- [x] Create a release tag on GitHub, from stable branch
- [x] Upload the new version to PyPI
- [x] Bump the version with "-dev" in the develop branch
- [x] Fill the release description on GitHub
- [x] Build conda packages
- [ ] Advertise the release on the mailing list
- [x] Close this issue

**Comment by ajoubertza** [comment]
@tiagocoutinho I don't see a final version of cppTango 9.3.x on [conda](https://anaconda.org/tango-controls/tango/files), but it is ticked off the list?

**Comment by tiagocoutinho** [comment]
We can ask the C++ team but I think 9.3.2alpha.1 is as good as we are going to have.
Future lib C++ 9.3.x must be ABI and API compatible so it should be ok for us.

**Comment by bourtemb** [comment]
Hi,

9.3.2alpha1 distrib is using cppTango 9.3.2.
You can refer to this [CHANGELOG file ](https://github.com/tango-controls/cppTango/blob/e84256df28f750fcaa26d41c7eceac9a5da968eb/CHANGELOG.md ) from tango-controls/cppTango#519 PR which will be merged just before the release, if you want to see the differences between cppTango 9.3.2 and cppTango 9.3.3.

**Comment by ajoubertza** [comment]
The major and minor version numbers for PyTango are supposed to follow the cppTango version.  At least that's what the [docs](https://github.com/tango-controls/pytango/blob/develop/doc/how-to-contribute.rst) say.  If we release PyTango 9.3.0, but there is no official 9.3.x release of cppTango that might be confusing for users.  Up to now cppTango only has "pre-release" versions at 9.3.x.

Do you think we should go ahead with a PyTango release anyway?  I expect it to work fine with cppTango 9.2.5.  It was just the event and polling unit tests that worked better with cppTango 9.3.2.

@bourtemb Do you have any idea when cppTango would get a final 9.3.x release?

**Comment by bourtemb** [comment]
I think we could already do a new cppTango 9.3.3 pre-release (just need to merge the CHANGELOG Pull Request).
We will install it in our control system at the ESRF and other institutes could test it too.
If there is no major bug found in the following weeks, we could flag it as stable release at that time.

**Comment by tiagocoutinho** [comment]
Thanks for clarifying @bourtemb!
Tango 9.3.2alpha comes inside with libtango 9.3.2. Since PyTango depends on libtango and not the entire tango package,I think we can go ahead and build a PyTango 9.3.0.

This makes me wonder @bourtemb : tango c++ conda package is kind of a hybrid: it contains libtango and docs but not the java tools. Should the tango c++ conda package version follow the tango package version or the libtango version?

**Comment by bourtemb** [comment]
> This makes me wonder @bourtemb : tango c++ conda package is kind of a hybrid: it contains libtango and docs but not the java tools. Should the tango c++ conda package version follow the tango package version or the libtango version?

According to your description, this Tango C++ conda package is linked to the C++ libtango version. 
So I think it would make sense for it to follow the cppTango version.

**Comment by ajoubertza** [comment]
OK, then lets go ahead with this 9.3.0 release.

**Comment by ajoubertza** [comment]
@tiagocoutinho Thanks for the work so far. When will you complete the final steps?  I notice the following
- on readthedocs "latest" still points to 9.2.5 instead of 9.3.0
- there are lots of files that still refer to the next release as 9.2.6 or 9.2.7 instead 9.3.0 or 9.3.1 - see all the changes that were made in PR #240.  Those files should be updated to reduce confusion.

**Comment by tiagocoutinho** [comment]
> on readthedocs "latest" still points to 9.2.5 instead of 9.3.0

Good catch. I just triggered the build. It should be ok now

> there are lots of files that still refer to the next release as 9.2.6 or 9.2.7 instead 9.3.0 or 9.3.1

you are right. I missed the README and 3 files in the doc. This is not critical but I agree it can lead to confusion. Since I deployed already on Pypi there is nothing I can do for 9.3.0. 

I propose we start using something like [bumpversion](https://github.com/peritus/bumpversion) to automatize a this process. It is quite boring and error prone.

I had problems with my user account on the mailing list. Fixing them now. I will make the announcement soon.
